subprojects/model-core/src/main/java/org/gradle/internal/instantiation/generator/AbstractClassGenerator.java

```diff
@@ -1159,6 +1165,24 @@ void applyTo(ClassGenerationVisitor visitor) {
         }
     }
 
+    private static class ConfigurableFileCollectionPropertyHandler extends ClassGenerationHandler {
+        private final List<PropertyMetadata> configurableFileCollectionTyped = new ArrayList<>();
+
+        @Override
+        void visitProperty(PropertyMetadata property) {
+            if (property.isReadOnly() && isConfigurableFileCollectionType(property.getType())) {
```

Comment:
❓ In `PropertyTypePropertyHandler` we are checking `Property.isReadable()`, here we are checking `isReadOnly()`, why is there a difference?

Reply:
Changed. I used readonly, since previously we generated a method that accepted `FileCollection`. So there was some issues if property was readable, since setter already existed. Now the method accepts `Object`, so there should be no more of this problem. 

Fixed.

Reply:
Actually reverted back to `readonly`. Otherwise the test:
```
FileCollectionIntegrationTest#can use += convenience in Groovy DSL to add elements to file collection when property has legacy setter
```
fails. 

Basically right now we are in a situation where we want to support `FileCollection +=` and not support `ConfigurableFileCollection +=`. And if we generate `setFromAnyValue` for any readable value we can't distinguish these two cases.

Reply:
Sounds good. Can you please add this as comment to the  code? The current condition is a bit hard to decipher without the additional context.